src/test/java/com/dreamsportslabs/guardian/it/ContactOtpVerifyIT.java

```diff
@@ -0,0 +1,416 @@
+package com.dreamsportslabs.guardian.it;
+
+import static com.dreamsportslabs.guardian.Constants.BODY_PARAM_CHANNEL;
+import static com.dreamsportslabs.guardian.Constants.BODY_PARAM_CONTACT;
+import static com.dreamsportslabs.guardian.Constants.BODY_PARAM_IDENTIFIER;
+import static com.dreamsportslabs.guardian.Constants.BODY_PARAM_OTP;
+import static com.dreamsportslabs.guardian.Constants.BODY_PARAM_STATE;
+import static com.dreamsportslabs.guardian.Constants.CODE;
+import static com.dreamsportslabs.guardian.Constants.ERROR;
+import static com.dreamsportslabs.guardian.Constants.ERROR_INCORRECT_OTP;
+import static com.dreamsportslabs.guardian.Constants.ERROR_INVALID_STATE;
+import static com.dreamsportslabs.guardian.Constants.ERROR_RETRIES_EXHAUSTED;
+import static com.dreamsportslabs.guardian.Constants.METADATA;
+import static com.dreamsportslabs.guardian.Constants.RESPONSE_BODY_PARAM_RETRIES_LEFT_METADATA;
+import static com.dreamsportslabs.guardian.Constants.RESPONSE_BODY_PARAM_STATE;
+import static com.dreamsportslabs.guardian.constant.Channel.SMS;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_NO_CONTENT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isA;
+
+import com.dreamsportslabs.guardian.utils.ApplicationIoUtils;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.stubbing.StubMapping;
+import io.restassured.response.Response;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class ContactOtpVerifyIT {
+  private static final String TENANT_ID = "tenant1"; // OTP is mocked for this tenant
+  private static final String TENANT_ID_NON_MOCKED = "tenant2"; // OTP is NOT mocked for this tenant
+  private WireMockServer wireMockServer;
+
+  private StubMapping getStubForSendSms() {
+    return wireMockServer.stubFor(
+        post(urlPathMatching("/sendSms"))
+            .willReturn(aResponse().withStatus(200).withBody("{\"result\":\"ok\"}")));
+  }
+
+  @Test
+  @DisplayName("Should verify OTP successfully with valid state and OTP (mocked tenant)")
+  public void testVerifyOtpSuccessfulMocked() {
+    // First send OTP to get state
```

Comment:
If you can add comments that indicate the 3 test phases for better readability, 
Setup, Execute, Validate. 
You can refer other Test files for the same